docs/adapters/configuration.md

```diff
@@ -0,0 +1,110 @@
+## Goal
```

Comment:
As a design discussion this is good but I think we should update docs/reference/cli.md to reflect what we ended up settling on and not include this doc in the final set of changes.